apps/wallet/src/ui/app/pages/dapp-tx-approval/index.tsx

```diff
@@ -80,17 +84,31 @@ function PassedObject({ id, module }: { id: string; module: string }) {
         TRUNCATE_MAX_LENGTH,
         TRUNCATE_PREFIX_LENGTH
     );
+
+    // Render the NFT if ID is a NFT
+    const nftMeta = useGetNFTMeta(id);
+
     return (
-        <div>
-            <ExplorerLink
-                type={ExplorerLinkType.object}
-                objectID={id}
-                className={st.objectId}
-                showIcon={false}
-            >
-                {objectId}
-            </ExplorerLink>
-            <div className={st.objectName}>{module}</div>
+        <div className={st.permissionsContent}>
+            <div className={st.permissionsContentLabel}>
+                <ExplorerLink
+                    type={ExplorerLinkType.object}
+                    objectID={id}
+                    className={cl(st.objectId, 'text-sui-dark')}
+                    showIcon={false}
+                >
+                    {objectId}
+                </ExplorerLink>
+                <div className={st.objectName}>{module}</div>
+            </div>
+
+            {nftMeta && nftMeta.url && (
```

Comment:
What is the case where there's NFTMeta but no URL? Seems like bad typings, as all NFTs must have a URL?